pkg/provisioner/crdprovisioner.go

```diff
@@ -315,17 +336,20 @@ func (c *CrdProvisioner) ExpandVolume(
 			return false, err
 		}
 		// Checking that the status is updated with the required capacityRange
-		if azVolumeUpdated != nil && azVolumeUpdated.Status != nil && azVolumeUpdated.Status.DeepCopy().ResponseObject.CapacityBytes == capacityRange.RequiredBytes {
+		if azVolumeUpdated.Status.ResponseObject.CapacityBytes == capacityRange.RequiredBytes {
```

Comment:
Might be good to leave the azVolumeUpdated.Status != nil check here?

Reply:
Ah, yes. Good point - we should be defensive against this.